app/src/org/commcare/adapters/JobListConnectHomeAppsAdapter.java

```diff
@@ -7,69 +7,107 @@
 import org.commcare.dalvik.databinding.ItemLoginConnectHomeCorruptAppsBinding;
+import org.commcare.dalvik.databinding.ItemLoginConnectSectionHeaderBinding;
 import org.commcare.interfaces.OnJobSelectionClick;
+import org.commcare.models.connect.ConnectListItem;
 import org.commcare.models.connect.ConnectLoginJobListModel;
 
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
 import java.util.Locale;
 
 public class JobListConnectHomeAppsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
 
     private Context mContext;
-    private final ArrayList<ConnectLoginJobListModel> jobList;
     private final OnJobSelectionClick launcher;
-    private final ArrayList<ConnectLoginJobListModel> corruptJobs;
-    private static final int NON_CORRUPT_JOB_VIEW = 1;
-    private static final int CORRUPT_JOB_VIEW = 2;
 
-    public JobListConnectHomeAppsAdapter(Context context, ArrayList<ConnectLoginJobListModel> jobList,
-            ArrayList<ConnectLoginJobListModel> corruptJobs, OnJobSelectionClick launcher) {
+    private final ArrayList<ConnectListItem> displayItems = new ArrayList<>();
+
+    private static final int VIEW_TYPE_SECTION_HEADER = 0;
+    private static final int VIEW_TYPE_CORRUPT_JOB = 1;
+    private static final int VIEW_TYPE_NON_CORRUPT_JOB = 2;
+
+    public JobListConnectHomeAppsAdapter(
+            Context context,
+            ArrayList<ConnectLoginJobListModel> inProgressJobs,
+            ArrayList<ConnectLoginJobListModel> newJobs,
+            ArrayList<ConnectLoginJobListModel> completedJobs,
+            ArrayList<ConnectLoginJobListModel> corruptJobs,
+            OnJobSelectionClick launcher
+    ) {
         this.mContext = context;
-        this.jobList = jobList;
-        this.corruptJobs = corruptJobs;
         this.launcher = launcher;
+        buildDisplayList(inProgressJobs, newJobs, completedJobs, corruptJobs);
     }
 
     @NonNull
     @Override
     public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
-        // Inflate the layout for each item using View Binding
-        if (viewType == NON_CORRUPT_JOB_VIEW) {
-            ItemLoginConnectHomeAppsBinding binding = ItemLoginConnectHomeAppsBinding.inflate(
-                    LayoutInflater.from(parent.getContext()), parent, false);
-            return new NonCorruptJobViewHolder(binding);
-        } else {
-            ItemLoginConnectHomeCorruptAppsBinding binding = ItemLoginConnectHomeCorruptAppsBinding.inflate(
-                    LayoutInflater.from(parent.getContext()), parent, false);
-            return new CorruptJobViewHolder(binding);
-        }
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+
+        return switch (viewType) {
+            case VIEW_TYPE_SECTION_HEADER -> new SectionHeaderViewHolder(
+                    ItemLoginConnectSectionHeaderBinding.inflate(inflater, parent, false)
```

Comment:
would be nice to rename these layouts to `job_list_section_header` and `job_list_corrupt_jobs_header` respectively as they don't really have anything do with `login`.